File: updatepromise.php

<?php

function majorityGate($a, $b, $c, $d) {
	$inputArray = array($a, $b, $c, $d);
	$count = 0;
	foreach ($inputArray as $value) {
		if( $value === true){
			$count += 1;
		}
	}
	 
	if ( $count > 2 ) {
		return true;
	}else {
		return false;
       }
}

function firstDuplicate($arr) {
	$len = count($arr);
	for( $i = 0; $i < $len-1; $i++) {
		for($j = $i+1; $j < $len; $j++) {
			if( $arr[$i] == $arr[$j] ) {
				return $j;
			}
		}
	}
	return -1;
}
    
function frenchWeeks($weeks) {
	if( $weeks <= 0 ) {
		return 0;
	}
	$rate = (7 * 24 * 60 * 60) / (10 * 10 * 100 * 100);
	return ($rate * $weeks);
}


echo "No1: Test Case for majorityGate <br/>";
$a = true;
$b = true;
$c = true;
$d = false;
if( majorityGate($a, $b, $c, $d) === true ) {
	echo "TRUE";
}else {
	echo "FALSE";
}
echo "<br/>";
echo "No2: Test Case for firstDuplicate <br/>";
$arr = array(0, 1, 2, 3, 4, 2, 6);
$index = firstDuplicate($arr);
echo $index;
echo "<br/>";

echo "No3: Test Case for frenchWeeks <br/>";
$weeks =  10.89;
$fweeks = frenchWeeks($weeks);
echo "frenchWeeks=" . $fweeks . "<br/>";
?>

File: updatepromise.cpp
#include <iostream>
#include <vector>
#include <set>
using namespace std;

bool majorityGate(bool a, bool b, bool c, bool d) {
    int count = 0;
    vector<bool> vt;
    vt.push_back(a);
    vt.push_back(b);
    vt.push_back(c);
    vt.push_back(d);
    for( auto it = vt.begin(); it != vt.end(); ++it) {
        if( *it ) {
            count++;
        }
    }

    if( count > 2 ) {
        return true;
    }else {
        return false;
    }
}

int firstDuplicate(vector<int> *vt) {
    set<int> mSet;
    pair<set<int>::iterator, bool> p;
    int index = 0;
    for( auto it = vt->begin(); it != vt->end(); ++it, ++index) {
        p = mSet.insert(*it);
        if( !(p.second) )  {
            return index;
        }
    }
    return -1;
}

float frenchWeeks( float weeks){
     if( weeks <= 0) {
        return 0;
    }
    float fweeks = 0.0;
    float rate = (float)(7 * 24 * 60 * 60) / (float)(10 * 10 * 100 * 100);
    return (weeks * rate); 
}

int main() {

    cout<<"No1: Test for majorityGate" <<endl;
    bool a = true;
    bool b = true;
    bool c = true;
    bool d = false;
    if( majorityGate(a, b, c, d)){
        cout << "TRUE" << endl;
    }else {
        cout << "FALSE" <<endl;
    }

    cout<<"No2: Test for firstDuplicate " <<endl;
    vector<int> nums = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 2};
    int index = firstDuplicate(&nums);
    cout << "index = " << index <<endl;

    cout<<"No3: Test for frenchWeeks " <<endl;
    float fweeks = frenchWeeks(10.89);
    cout<<"frenchWeeks= " << fweeks <<endl;

    return 0;
}

